src/compiler/binder.rs

```diff
@@ -421,7 +420,8 @@ impl BinderType {
         name: __String,
     ) -> Rc<Symbol> {
         let symbol = self.create_symbol(symbol_flags, name).wrap();
-        enum_unwrapped!(&*self.file(), [Node, SourceFile])
+        self.file()
+            .as_source_file()
```

Comment:
The advantages of `.as_*()` are that (a) you don't need the `&*` boilerplate for `Rc`'s because they can deref to the `.as_*()` method calls on their inner `Node`/`Type` and (b) per one comment on previous PR, this puts things in more of a "pipeline" order

Reply:
Nice, this reads really nicely, even aside from the boilerplate reduction.